client/src/js/directives/findDebtorGroup.js

```diff
@@ -0,0 +1,116 @@
+/**
+* <bh-find-debtorgroup></bh-find-debtorgroup>
+*
+* The find debtor group component is responsible for displaying a component
+* for searching a debtor group or a convention
+*
+* This component takes these attributes :
+*   - locked : if we want only debtor groups which are locked or not
+*   - is-convention : if we want only debtor groups which are conventions or not
+*   - on-search-complete : the callback function which get the returned debtror group
+*
+* @example
+* // component find debtor group
+* <bh-find-debtorgroup>
+*   locked="true"
+*   is-convention="true"
+*   on-search-complete="MyCtrl.handleResult"
+* </bh-find-debtorgroup>
+*/
+
+/** Find Debtor Group Controller */
+function findDebtorGroupController($http, DebtorGroupService) {
+  var ctrl         = this,
+      callback     = this.callback,
+      locked       = this.locked,
```

Comment:
Can `locked` or `isConvention` change dynamically?   I think that copying these values to a local variable will eliminate that possibility.  For example,
```js
var o = { x : 1 };
var alias = o.x;
console.log(o.x); //  1
console.log(alias); // 1
o.x = 2;
console.log(o.x); // 2
console.log(alias) // 1
```
The local copy (`alias`) does not change, but the reference (`o.x`) does change.